src/nupic/math/ArrayAlgo.hpp

```diff
@@ -3779,7 +3779,7 @@ namespace nupic {
       // n1 is the number floats we can process in parallel with xmm
       // n2 is the number of "stragglers" what we will have to do one by one ( < 4)
       nupic::Real32 count = 0;
-      long x_addr = (long) begin; // 8 bytes on 64 bits platforms
+      long long x_addr = (long long) begin; // 8 bytes on 64 bits platforms
```

Comment:
Yes, uintptr_t looks like what we want but doesn't look like it will necessarily be portable. We can add NTA_UIntPtr to Types.h and a typedef for UIntPtr in Types.hpp. If some platforms don't support uintptr_t then we can use preprocessor checks in those files.